aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.kt

```diff
@@ -684,7 +687,56 @@ class AWSCognitoAuthPlugin : AuthPlugin<AWSCognitoAuthServiceBehavior>() {
         onSuccess: Action,
         onError: Consumer<AuthException>
     ) {
-        TODO("Not yet implemented")
+        var listenerToken: StateChangeListenerToken? = null
+        listenerToken = credentialStoreStateMachine.listen(
+            {
+                when (it) {
+                    is CredentialStoreState.Success -> {
+                        listenerToken?.let(credentialStoreStateMachine::cancel)
+                        if (it.storedCredentials?.cognitoUserPoolTokens?.accessToken != null) {
+                            _updatePassword(it.storedCredentials.cognitoUserPoolTokens.accessToken, oldPassword, newPassword, onSuccess, onError)
+                        } else {
+                            onError.accept(AuthException.InvalidAccountTypeException())
+                        }
+                    }
+                    is CredentialStoreState.Error -> {
+                        listenerToken?.let(credentialStoreStateMachine::cancel)
+                        onError.accept(AuthException.UnknownException(it.error))
+                    }
+                    else -> {
+
+                    }
+                }
+            },
+            {
+                credentialStoreStateMachine.send(
+                    CredentialStoreEvent(CredentialStoreEvent.EventType.LoadCredentialStore())
+                )
+            }
+        )
+    }
+
+    private fun _updatePassword(token: String,
+                                oldPassword: String,
+                                newPassword: String,
+                                onSuccess: Action,
+                                onError: Consumer<AuthException>) {
+        val changePasswordRequest = ChangePasswordRequest.invoke {
+            previousPassword = oldPassword
+            proposedPassword = newPassword
+            accessToken = token
+        }
+
+        try {
+            GlobalScope.launch {
```

Comment:
Check if user is signed in by checking the current state of AuthStateMachine using `getCurrentState`.